site/app/templates/officeHoursQueue/QueueStatus.twig

```diff
@@ -96,6 +96,14 @@
       <input type="hidden" name="queue_code" value="{{ viewer.getCurrentQueueCode() }}"/>
       <button id="leave_queue" type="submit" class="btn btn-danger">Leave Queue</button>
     </form>
+      {% set my_queue = 'code' %}
+      {% set open_queues = viewer.getAllOpenQueues()%}
+      {% if open_queues|length == 1 %}
+          {% for queue in viewer.getAllOpenQueues() %}
+            {% set my_queue = queue['code'] %}
+          {% endfor %}
+      {% endif %}
+      {% if open_queues|length > 1  or my_queue != viewer.getCurrentQueueCode() %}
```

Comment:
I don't think the twig code above is needed. But I don't fully understand what its trying to do,
it looks like if there's exactly one open queue then set `my_queue` to the last open queue's code? That doesn't seem right - I'm guessing the `== 1` should be a `> 1`

You could do something like instead
```php
{% if open_queues|length > 1 or (viewer.getAllOpenqueues() | last)['code'] != viewer.getAllOpenQueues()) %}
```

If this if block is supposed to render the switch queues form, wouldn't the logic just be 
`if open_queues|length > 1` ? why do you need to compare queue codes

Reply:
There is an edge case where the student is in queue A, but we close that queue and open queue B.  They should be able to switch to queue B, even if there is only 1 queue option.  (So testing if there is <= 1 queues open is not the right check in this case).

Reply:
ok